verifier/verifier.go

```diff
@@ -63,6 +66,25 @@ func New(trustPolicy *trustpolicy.Document, trustStore truststore.X509TrustStore
 	}, nil
 }
 
+// NewWithRevocationClient creates a new verifier given trustPolicy, trustStore, pluginManager, and revocationClient
+func NewWithRevocationClient(trustPolicy *trustpolicy.Document, trustStore truststore.X509TrustStore, pluginManager plugin.Manager, revocationClient *http.Client) (notation.Verifier, error) {
+	if revocationClient == nil {
+		revocationClient = &http.Client{Timeout: 5 * time.Second}
+	}
+	result, err := New(trustPolicy, trustStore, pluginManager)
+	if err != nil {
+		return nil, err
+	}
+
+	v, ok := result.(*verifier)
+	if !ok {
+		// should never occur, but checking just in case
+		return nil, errors.New("unable to create verifier with client for revocation")
+	}
```

Comment:
Why do we need to do this? New should always return `notation.Verifier`

Reply:
Exactly. It returns notation.Verifier, but the private revocationClient is not defined for this struct, only the local verifier struct includes this (which we know is returned, but the function declares the public interface). Thus, we need to cast from notation.Verifier to verifier. This has been discussed previously in https://github.com/notaryproject/notation-go/pull/295#discussion_r1151292505